src/main/java/com/test/scrapers/JsoupScraper.java

```diff
@@ -0,0 +1,163 @@
+            return;
+        }
+
+        String url;
+        if (path.contains("://")) {
+            URI pathUri = new URI(path);
+            if (!pathUri.getHost().contains(rootUri.getHost())) {
+                // don't follow links off site
+                return;
+            }
+            url = path;
+        } else {
+            url = rootUri.toString();
+            if (!path.startsWith("/")) {
+                url += "/";
+            }
+            url += path;
+        }
+
+        // strips off protocol header
+        URI uri = new URI(url);
+        String localPath = uri.getSchemeSpecificPart() + uri.getRawPath();
+        if ((localPath.length() > 0) && visitedUrls.contains(localPath)) {
+            // prevent circular loops
+            return;
+        } else {
+            visitedUrls.add(localPath);
+        }
+        logger.debug("Scraping " + url);
+        System.out.print(".");  // progress indicator
+
+        Document doc = useWebClient ?
+                Jsoup.parse(loadPageSelenium(url))
+                : Jsoup.connect(url).get();
+
+        logger.debug("Document is " + doc.outerHtml());
+
+        // extract emails
+        Matcher matcher = emailRegEx.matcher(doc.text());
+        while (matcher.find()) {
+            String email = matcher.group();
+            emails.add(email);
+
+            logger.info("Found email: " + email);
+            System.out.print("+");  // progress indicator
+        }
+
+        // now walk all the links on the page recursively
+        Elements elements = doc.select("a[href]");
+        for (Element e : elements) {
+            String childUrl = e.attr("href");
+            if (childUrl != null) {
+                try {
+                    scrapeEmailsFromPage(rootUri, childUrl);
+                } catch (IOException | URISyntaxException ex) {
+                    // just log exceptions
+                    logger.warn("Error scraping '" + childUrl + "': " + ex);
+                }
+            }
+        }
+    }
+
+    private String loadPageSelenium(String url) {
+        // this can throw a lot of different exceptions
+
+        // disable SSL checking (set to "compatible" for Firefox SSL certs)
+        Properties props = System.getProperties();
+        props.setProperty("jbd.pemfile", "trustanything");
+
+        JBrowserDriver driver = new JBrowserDriver(Settings.builder().
+                timezone(Timezone.AMERICA_NEWYORK).build());
+
+        // This will block for the page load and any
+        // associated AJAX requests
+        driver.get(url);
+
+        // ignore pages w/ errors
+        if (driver.getStatusCode() != 200) {
+            return "";
+        }
+
+        // Returns the page source in its current state, including
+        // any DOM updates that occurred after page load
+        String result = driver.getPageSource();
+
+        // Close the browser. Allows the Selenium thread to terminate.
+        driver.quit();
```

Comment:
yes...thought it was safer in case it got into a bad state.  I didn't see much difference in speed...I did switch it over so it doesn't keep recreating the driver.